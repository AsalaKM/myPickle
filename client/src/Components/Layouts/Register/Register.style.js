import styled from "styled-components"

const Intro = styled.div.attrs({
  className: "flex flex-column items-center ph3",
})`
  padding-top: 2rem;
`

const Button = styled.button.attrs({
  className: "mp-bg-white mp-b--primary-color",
})`
  border-style: solid;
  border-width: 1px;
  border-radius: 2rem;
  cursor: pointer;
  margin-top: 2rem;
  width: 40vw;
  height: 3rem;

  :hover {
    background-color: var(--primary);
    color: var(--white);
  }
  :focus {
    background-color: var(--darkPrimary);
    color: var(--white);
  }
`

const CircleLarge = styled.div.attrs({
  className: "mp-b--primary-color",
})`
  border-style: solid;
  position: absolute;
  border-radius: 50%;
  border-width: 1px;
  left: -40vw;
  width: 80vw;
  height: 80vw;
`

const CircleMed = styled(CircleLarge)`
  width: 60vw;
  height: 60vw;
  left: -30vw;
  top: 10vw;
`

const CircleSm = styled(CircleLarge)`
  width: 40vw;
  height: 40vw;
  left: -20vw;
  top: 20vw;
`

const CirclesContainer = styled.div`
  position: relative;
  height: 80vw;
`

export { Intro, Button, CircleLarge, CircleMed, CirclesContainer, CircleSm }
