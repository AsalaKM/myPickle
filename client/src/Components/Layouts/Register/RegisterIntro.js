import React from "react"

import { Intro } from "../../Common/Headings"
import { Button } from "../../Common/Buttons"

export default class RegisterIntro extends React.Component {
  render() {
    return (
      <Intro>
        <h2 className="tc mp-primary-color">Welcome to myPickle</h2>
        <p>
          Please provide your contact and organisation details first. Then you can begin to list the
          service(s) you provide.
        </p>
        <p>
          If at any point you need assistance, please email{" "}
          <a href="mailto:hello@mypickle.org">hello@mypickle.org</a> with your query and contact
          details.
        </p>
        <Button onClick={this.props.handleNext}>Next</Button>
      </Intro>
    )
  }
}
