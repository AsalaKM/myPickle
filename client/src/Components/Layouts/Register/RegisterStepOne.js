import React from "react"

import { Intro, Button } from "./Register.style"

export default class RegisterIntro extends React.Component {
  render() {
    console.log(this.props)
    return (
      <Intro>
        <h2 className="tc mp-primary-color">Wellness Areas</h2>
        <p>
          So we can customize your profile, please select the area(s) of wellness where you provide
          support.
        </p>
        <div className="flex items-center justify-between w-100">
          <Button onClick={this.props.handlePrevious}>Back</Button>
          <Button onClick={this.props.handleNext}>Next</Button>
        </div>
      </Intro>
    )
  }
}
