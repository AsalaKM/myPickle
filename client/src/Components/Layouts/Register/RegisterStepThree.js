import React from "react"

import { Button } from "./Register.style"
import TextInput from "../../Common/Questions/TextInput"
import RadioInput from "../../Common/Questions/RadioInput"
import FileUploadInput from "../../Common/Questions/FileUploadInput"

export default class RegisterStepThree extends React.Component {
  render() {
    const {
      handleChange,
      answers,
      basicInfoQuestions,
      handleSubmit,
      unanswered,
      checkRequiredAnswers,
      imageUpload,
    } = this.props

    console.log("STEP3", basicInfoQuestions)
    return (
      <React.Fragment>
        <div className="BasicInfo">
          {basicInfoQuestions.map(question => {
            // pull out the details from each question
            const { inputType } = question

            if (inputType === "text" || inputType === "url") {
              return (
                <TextInput
                  question={question}
                  handleChange={handleChange}
                  answers={answers}
                  unanswered={unanswered}
                  checkRequiredAnswers={checkRequiredAnswers}
                />
              )
            }
            if (inputType === "radio") {
              return (
                <RadioInput
                  question={question}
                  handleChange={handleChange}
                  answers={answers}
                  unanswered={unanswered}
                  checkRequiredAnswers={checkRequiredAnswers}
                />
              )
            }
            if (inputType === "file-upload") {
              return (
                <FileUploadInput
                  question={question}
                  handleChange={handleChange}
                  answers={answers}
                  unanswered={unanswered}
                  checkRequiredAnswers={checkRequiredAnswers}
                  imageUpload={imageUpload}
                />
              )
            }
          })}
        </div>
        <div className="flex items-center justify-between w-100">
          <Button id="prev-btn" onClick={this.props.handlePrevious}>
            Back
          </Button>
          <Button className="submit" onClick={handleSubmit}>
            Submit
          </Button>
        </div>
      </React.Fragment>
    )
  }
}
