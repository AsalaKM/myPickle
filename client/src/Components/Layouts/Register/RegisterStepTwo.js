import React from "react"

import { Button, RegisterStepTwoWrapper } from "./Register.style"
import { TextField } from "../../Common/Questions/Questions.style"
import TextInput from "../../Common/Questions/TextInput"
import TextFieldInput from "../../Common/Questions/TextFieldInput"

export default class RegisterStepTwo extends React.Component {
  render() {
    const {
      handleChange,
      answers,
      adminQuestions,
      checkRequiredAnswers,
      unanswered,
      checkStage,
    } = this.props
    return (
      <RegisterStepTwoWrapper>
        <TextField>
          <header>
            <h4>Contact Name:*</h4>
            <p />
          </header>
          <input
            type="text"
            name="name"
            onChange={handleChange}
            value={answers["name"] ? answers["name"] : ""}
            onBlur={checkRequiredAnswers}
            required
          />
          {unanswered && unanswered.includes("name") ? (
            <div className="required">
              <p>Please answer this question</p>
            </div>
          ) : (
            ""
          )}
        </TextField>
        <TextField>
          <header>
            <h4>Contact Email:*</h4>
            <p />
          </header>
          <input
            type="email"
            name="email"
            onChange={handleChange}
            value={answers["email"] ? answers["email"] : ""}
            onBlur={checkRequiredAnswers}
          />
          {unanswered && unanswered.includes("email") ? (
            <div className="required">
              <p>Please answer this question</p>
            </div>
          ) : (
            ""
          )}
        </TextField>
        <TextField>
          <header>
            <h4>Contact Password:*</h4>
            <p />
          </header>
          <input
            type="password"
            name="password"
            onChange={handleChange}
            value={answers["password"] ? answers["password"] : ""}
            onBlur={checkRequiredAnswers}
          />
          {unanswered && unanswered.includes("password") ? (
            <div className="required">
              <p>Please answer this question</p>
            </div>
          ) : (
            ""
          )}
        </TextField>
        <TextField>
          <header>
            <h4>Confirm Password:*</h4>
            <p />
          </header>
          <input
            type="password"
            name="password2"
            onChange={handleChange}
            value={answers["password2"] ? answers["password2"] : ""}
            onBlur={checkRequiredAnswers}
          />
          {unanswered && unanswered.includes("password2") ? (
            <div className="required">
              <p>Please answer this question</p>
            </div>
          ) : (
            ""
          )}
        </TextField>
        <TextField>
          <header>
            <h4>Phone Number:*</h4>
            <p />
          </header>
          <input
            type="text"
            name="phone"
            onChange={handleChange}
            value={answers["phone"] ? answers["phone"] : ""}
            onBlur={checkRequiredAnswers}
          />
          {unanswered && unanswered.includes("phone") ? (
            <div className="required">
              <p>Please answer this question</p>
            </div>
          ) : (
            ""
          )}
        </TextField>
        {adminQuestions.map(question => {
          const { inputType } = question

          if (inputType === "text") {
            return (
              <TextInput
                question={question}
                handleChange={handleChange}
                answers={answers}
                unanswered={unanswered}
                checkRequiredAnswers={checkRequiredAnswers}
              />
            )
          }
          if (inputType === "textarea" || inputType === "address") {
            return (
              <TextFieldInput
                question={question}
                handleChange={handleChange}
                answers={answers}
                unanswered={unanswered}
                checkRequiredAnswers={checkRequiredAnswers}
              />
            )
          }
        })}
        <div className="flex items-center justify-between w-100 mb4">
          <Button id="prev-btn" onClick={this.props.handlePrevious}>
            Back
          </Button>
          <Button className="next-btn" onClick={checkStage}>
            Next
          </Button>
        </div>
      </RegisterStepTwoWrapper>
    )
  }
}
