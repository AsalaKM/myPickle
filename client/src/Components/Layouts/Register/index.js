import React, { Component } from "react"

import RegisterIntro from "./RegisterIntro"
import RegisterStepOne from "./RegisterStepOne"
import RegisterStepTwo from "./RegisterStepTwo"

import axios from "axios"

class Register extends Component {
  state = {
    registerQuestions: null,
    registerAnswers: {},
    position: 0,
    unanswered: [],
  }

  componentDidMount() {
    axios
      .get("/get-register-questions")
      .then(res => this.setState({ registerQuestions: res.data }))
      .catch(err => console.log("message", err))
  }

  checkRequiredAnswers = question => {
    const value = question.target.value
    const questionId = question.target.name
    const newUnanswered = this.state.unanswered
    if (!value) {
      newUnanswered.push(questionId)
    } else {
      if (newUnanswered.includes(questionId)) {
        const index = newUnanswered.indexOf(questionId)
        newUnanswered.splice(index, 1)
      }
    }

    this.setState({ unanswered: newUnanswered })
  }

  checkStage = () => {
    const answerState = this.state.registerAnswers
    const newUnanswered = this.state.unanswered
    let counter = 0

    if (this.state.position === 1) {
      const wellnessQuestion = this.state.registerQuestions[0]
      if (!answerState[wellnessQuestion._id] || answerState[wellnessQuestion._id].length < 1) {
        if (!newUnanswered.includes(wellnessQuestion._id)) newUnanswered.push(wellnessQuestion._id)
        counter += 1
        console.log(newUnanswered)
      } else {
        if (newUnanswered.includes(wellnessQuestion._id)) {
          const index = newUnanswered.indexOf(wellnessQuestion._id)
          newUnanswered.splice(index, 1)
        }
      }
    }
    if (this.state.position === 2) {
      const array = ["name", "email", "password", "password2", "phone"]
      array.forEach(item => {
        if (!answerState[item] || answerState[item].length < 1) {
          if (!newUnanswered.includes(item)) newUnanswered.push(item)
          counter += 1
          console.log(newUnanswered)
        }
      })
    }
    if (counter === 0) {
      this.handleNext()
    } else this.setState({ unanswered: newUnanswered })
  }

  handleChange = option => {
    // set up id of input field as the name attribute of that input
    const questionId = option.target.name
    const newAnswerState = this.state.registerAnswers
    const newUnanswered = this.state.unanswered
    let answer

    if (option.target.type === "checkbox") {
      if (newUnanswered.includes(questionId)) {
        const index = newUnanswered.indexOf(questionId)
        newUnanswered.splice(index, 1)
      }
      answer = option.target.value
      if (!newAnswerState[questionId]) {
        newAnswerState[questionId] = [answer]
      } else if (!newAnswerState[questionId].includes(answer)) {
        newAnswerState[questionId].push(answer)
      } else if (newAnswerState[questionId].includes(answer)) {
        const index = newAnswerState[questionId].indexOf(answer)
        newAnswerState[questionId].splice(index, 1)
      }
    } else {
      answer = option.target.value
      newAnswerState[questionId] = answer
    }
    this.setState({ registerAnswers: newAnswerState, unanswered: newUnanswered })
  }

  handleNext = () => {
    this.setState({ position: this.state.position + 1 })
  }

  handlePrevious = e => {
    e.preventDefault()
    this.setState({ position: this.state.position - 1 })
  }

  render() {
    const { position } = this.state
    if (position === 0) {
      return (
        <React.Fragment>
          <RegisterIntro handleNext={this.handleNext} />
        </React.Fragment>
      )
    }
    if (position === 1) {
      return (
        <React.Fragment>
          <RegisterStepOne
            handleNext={this.handleNext}
            handlePrevious={this.handlePrevious}
            wellnessQuestion={this.state.registerQuestions[0]}
            answers={this.state.registerAnswers}
            handleChange={this.handleChange}
            checkStage={this.checkStage}
            unanswered={this.state.unanswered}
          />
        </React.Fragment>
      )
    }
    if (position === 2) {
      return (
        <React.Fragment>
          <RegisterStepTwo
            handleNext={this.handleNext}
            handlePrevious={this.handlePrevious}
            handleChange={this.handleChange}
            answers={this.state.registerAnswers}
            checkRequiredAnswers={this.checkRequiredAnswers}
            unanswered={this.state.unanswered}
            checkStage={this.checkStage}
          />
        </React.Fragment>
      )
    }
    if (position === 3) {
      return (
        <React.Fragment>
          <h4>Hello!</h4>
        </React.Fragment>
      )
    }
  }
}

export default Register
