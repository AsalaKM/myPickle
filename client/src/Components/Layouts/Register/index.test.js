import React from "react"
// import renderer from "react-test-renderer"
// import { StaticRouter } from "react-router-dom"
import { shallow, mount } from "enzyme"

import Register from "./index"
import RegisterStepOne from "./RegisterStepOne"

const mockUpdatePosition = jest.fn()

describe("Register component", () => {
  test("should render correctly", () => {
    const component = shallow(<Register />)
    expect(component).toMatchSnapshot()
  })

  test("check position one renders", () => {
    const component = shallow(<Register />)
    // const context = {}
    // const innerComponent = shallow(<RegisterStepOne />)
    component.setState({ position: 1 })

    const newComponent = shallow(
      <Register>
        <RegisterStepOne />
      </Register>
    )
    expect(newComponent).toMatchSnapshot()
  })

  test("check position two renders", () => {
    const component = shallow(<Register />)
    // const context = {}
    // const innerComponent = shallow(<RegisterStepOne />)
    component.setState({ position: 2 })

    const newComponent = shallow(
      <Register>
        <h1>Great progress!</h1>
      </Register>
    )
    expect(newComponent).toMatchSnapshot()
  })

  // test("handleNext function", () => {
  //   const component = mount(<Register />)
  //   const innerComp = shallow(<RegisterStepOne />)

  //   console.log(innerComp.debug())

  //   console.log("BEFORE", component.state().position)

  //   innerComp.find(".next-btn").simulate("click")

  //   console.log("AFTER", component.state().position)

  //   const newComponent = shallow(
  //     <Register>
  //       <RegisterStepOne />
  //     </Register>
  //   )
  //   expect(newComponent).toMatchSnapshot()

  //   // expect(component.state().position).toEqual(1)
  // })
})
