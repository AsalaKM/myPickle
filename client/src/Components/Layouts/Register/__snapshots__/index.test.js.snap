// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component check position one renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register>
    <RegisterIntro />
  </Register>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <RegisterIntro
        handleNext={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "handleNext": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <RegisterIntro
          handleNext={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleNext": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Register component check position two renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register>
    <h1>
      Great progress!
    </h1>
  </Register>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <RegisterIntro
        handleNext={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "handleNext": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <RegisterIntro
          handleNext={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleNext": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Register component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <RegisterIntro
        handleNext={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "handleNext": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <RegisterIntro
          handleNext={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleNext": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
